### PR TITLE
Fix: more clearer unexpected statement end error messages

### LINKED ISSUE #36811: Extends keyword prevents eol or semicolon from being found
**Godot version:**
3.1.2 Stable

**OS/device including version:**
Laptop
Linux Mint 19.1 Cinnamon
Cinnamon Version 4.0.10
Linux Kernel 4.15.0-20-generic
Intel Core i3-7100U

**Issue description:**
I am trying to use the following segment of code

```
if variable extends Array:
	pass
```
I receive an error saying  ": expected at end of line"

if i try doing the following

```
var condition = self extends Node2D
if condition:
	pass
``` 
 I receive an error saying "expected end of statement (var)"

**Steps to reproduce:**

To reproduce this error use the following code
```
var _dummy = self extends Node
```

**Minimal reproduction project:**
I have posted a project containing the error on Github [here](https://github.com/lolzyman/Extends_Error_Project)

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #36811

error-message-unexpected-eos

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -2888,7 +2888,7 @@ void GDScriptParser::_parse_block(BlockNode *p_block, bool p_static) {
 				lv->assign = assigned;
 
 				if (!_end_statement()) {
-					_set_error("Expected end of statement (\"var\").");
+					_set_error(String() + "Expected end of statement (\"var\"), got " + tokenizer->get_token_name(tokenizer->get_token()) + ":" + tokenizer->get_token_identifier() + " instead");
```
Comment (akien-mga) [inline modules/gdscript/gdscript_parser.cpp]: You can use `vformat()` instead:
```
vformat("Expected end of statement (\"var\"), got %s:%s instead.", tokenizer->get_token_name(tokenizer->get_token()), tokenizer->get_token_identifier())
```

Also do not remove the final dot, error messages are proper sentences that end with a dot.

Comment (akien-mga) [reply modules/gdscript/gdscript_parser.cpp]: What would be `tokenizer->get_token_identifier()` in this context? In your screenshot it seems to be an empty string.

Comment (ThakeeNathees) [reply modules/gdscript/gdscript_parser.cpp]: `var my_var = self undefined_idf`
`tokenizer->get_token()` prints `Identifier`
`tokenizer->get_token_identifier()` prints `undefined_idf`
but most tokens doesn't have identifier

I just copied:
https://github.com/godotengine/godot/blob/1500e40ccf88b0bca34667e3113e4d6c18b3de29/modules/gdscript/gdscript_parser.cpp#L5232

Comment (ThakeeNathees) [reply modules/gdscript/gdscript_parser.cpp]: error-message

Comment (ThakeeNathees) [reply modules/gdscript/gdscript_parser.cpp]: any changes in format for the error message ?
@akien-mga

Comment (ThakeeNathees) [reply modules/gdscript/gdscript_parser.cpp]: any changes in format for the error message ?
@akien-mga

Comment (akien-mga) [reply modules/gdscript/gdscript_parser.cpp]: Seems good :)

**Review by vnen — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs a rebase to solve a merge conflict.

**Comment by ThakeeNathees** [comment]
done rebase
@akien-mga

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**